data/base/endpoint.php

```diff
@@ -285,22 +285,30 @@ public function delete_items( $request ) {
 	/**
 	 * Register item route.
 	 *
+	 * @param string $methods
 	 * @param array $args
 	 * @param string $route
-	 * @param string $methods
+	 * @param bool $should_add_id_arg
 	 *
 	 * @throws \Exception
 	 */
-	public function register_item_route( $methods = WP_REST_Server::READABLE, $args = [], $route = '/' ) {
-		$args = array_merge( [
-			'id' => [
-				'description' => 'Unique identifier for the object.',
-				'type' => 'string',
-			],
-		], $args );
-
-		if ( isset( $args['id'] ) && $args['id'] ) {
-			$route .= '(?P<id>[\w]+)/';
+	public function register_item_route(
+		$methods = WP_REST_Server::READABLE,
+		$args = [],
+		$route = '/',
+		$should_add_id_arg = true
+	) {
+		if ( $should_add_id_arg ) {
+			$args = array_merge( [
```

Comment:
I think its better to check if $args is empty then add id if not use $args as it.